src/main/java/org/fungover/thunder/PackageReader.java

```diff
@@ -62,9 +63,18 @@ public boolean isCleanDisconnect(Socket socket) throws IOException {
 
         InputStream inputStream = socket.getInputStream();
         byte[] buffer = new byte[1024];
-        int bytesRead = inputStream.read(buffer);
+        int bytesRead = -1;
+
+        try {
+            socket.setSoTimeout(CONNECTION_TIMEOUT);
+            bytesRead = inputStream.read(buffer);
+        } catch (SocketTimeoutException e) {
+            System.err.println("Socket read timed out.");
+        } finally {
+            socket.setSoTimeout(0);
+        }
```

Comment:
You don't need to setSoTimeout(CONNECTION_TIMEOUT) in isCleanDisconnect. It should only timeout if it does not get the CONNECT packet in 30000ms.
Shouldn't putting setSoTimeout(0) after the broker has recieved the CONNECT packet be enough to fix this bug?
![image](https://github.com/fungover/thunder/assets/143023413/123ad560-2018-4fb7-88f1-c9a07243222e)